sputnikvm/src/vm/transaction.rs

```diff
@@ -77,22 +77,27 @@ impl Transaction {
             Transaction::MessageCall {
                 address, caller, gas_price, gas_limit, value, data
             } => {
+                account_state.require(caller)?;
+                account_state.require_code(address)?;
+
                 if !is_code {
-                    let nonce = account_state.nonce(caller)?;
+                    let nonce = account_state.nonce(caller).unwrap();
```

Comment:
The `panic!()` of `unwrap` is never expected to happen. There're two `require`/`require_code` statement at [line 80](https://github.com/ethereumproject/sputnikvm/pull/197/files/ed0f81c92077469b3c218cd72f6144f30c1e61c6#diff-ba1901e4539c16d0cd8966aba55c81e8R80). If `require` doesn't return an error, the `AccountState` ensures that calling `nonce`, `code`, etc with the given address would always return `Ok(..)`.

The problem with using `try!`/`?` in here is that after checking that the current situation, we're already starting to modify the VM's state. So it cannot return early -- otherwise the VM's state will be invalid. That's also why here it's changed from `?` to `unwrap()` and use `require` above. 😃